// @maskito/angular/esm2020/lib/deprecated.mjs
import { Directive, Input, NgModule } from '@angular/core';
import { MaskitoDirective } from './maskito.directive';
import { MaskitoPipe } from './maskito.pipe';
import * as i0 from "@angular/core";
/**
 * @deprecated Just use `MaskitoDirective`, `MaskitoCVA` no longer needed.
 */
export class MaskitoCVA {
}
MaskitoCVA.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.10", ngImport: i0, type: MaskitoCVA, deps: [], target: i0.ɵɵFactoryTarget.Directive });
MaskitoCVA.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "15.2.10", type: MaskitoCVA, isStandalone: true, selector: "input[maskito], textarea[maskito]", inputs: { maskito: "maskito" }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.10", ngImport: i0, type: MaskitoCVA, decorators: [{
            type: Directive,
            args: [{
                    standalone: true,
                    selector: 'input[maskito], textarea[maskito]',
                }]
        }], propDecorators: { maskito: [{
                type: Input
            }] } });
/**
 * @deprecated Use standalone `MaskitoDirective` and `MaskitoPipe` instead.
 * Learn more: https://maskito.dev/frameworks/angular
 * ___
 * TODO: Delete it in v3.0 (after Taiga UI 4.0 will be released and bumped in this repository).
 * Taiga UI (3.x.x) uses Maskito (1.x.x) <=> Maskito (2.x.x) uses Taiga UI (3.x.x) to build demo application.
 * Also, NPM workspaces create symlinks for `@maskito/*` inside `node_modules`.
 * All this together don't allow us to build Maskito's demo without this legacy module.
 * ___
 * > nx build demo
 * ```
 * ./node_modules/@taiga-ui/kit/fesm2015/taiga-ui-kit-components-input-date.js:465:29-42 -
 * Error: export 'MaskitoModule' (imported as 'MaskitoModule') was not found in '@maskito/angular'
 * (possible exports: MaskitoCVA, MaskitoDirective, MaskitoPipe)
 * ```
 */
export class MaskitoModule {
}
MaskitoModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.10", ngImport: i0, type: MaskitoModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
MaskitoModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.2.10", ngImport: i0, type: MaskitoModule, imports: [MaskitoDirective, MaskitoCVA, MaskitoPipe], exports: [MaskitoDirective, MaskitoCVA, MaskitoPipe] });
MaskitoModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.2.10", ngImport: i0, type: MaskitoModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.10", ngImport: i0, type: MaskitoModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [MaskitoDirective, MaskitoCVA, MaskitoPipe],
                    exports: [MaskitoDirective, MaskitoCVA, MaskitoPipe],
                }]
        }] });
